### PR TITLE
Add PasswordHasher and code to handle salt

### PR DESCRIPTION
**Pull request by erpmario**
Additions/modifications proposed in this pull request:
- Add `PasswordHasher`, a class that can generate salts and hash salted passwords with the `Sha256` algorithm.
- Modify the account creation, storage, and authentication code to compensate for the new `salt` field in the databases.
- Fix some minor formatting issues.

### CODE REVIEW

File: Projects/CoX/Utilities/dbtool/main.cpp [hunk 1]
```diff
@@ -131,6 +133,42 @@ void createDatabases()
     }
 }
 
+void addAdminAccount()
+{
```
Comment (nemerle) [inline Projects/CoX/Utilities/dbtool/main.cpp]: change the name of this to addAcount, add username, password and access_level as parameters, and we have generic method :)

File: Projects/CoX/Utilities/dbtool/main.cpp [hunk 2]
```diff
@@ -190,7 +228,7 @@ int main(int argc, char **argv)
         qWarning() << "Forced flag used '-f'. Existing databases may be overwritten.";
 
     createDatabases();
-
+    addAdminAccount();
```
Comment (nemerle) [inline Projects/CoX/Utilities/dbtool/main.cpp]:
```cpp
addAccount("segsadmin","segs123",9);
```

File: Projects/CoX/Utilities/dbtool/main.cpp [hunk 3]
```diff
@@ -92,7 +94,7 @@ void createDatabases()
         }
         QFile source_file(source_file_string);
         QFile target_file(target_file_string);
-        QSqlDatabase segs_db(QSqlDatabase::addDatabase("QSQLITE"));
+        QSqlDatabase segs_db = QSqlDatabase::addDatabase("QSQLITE");
         QSqlQuery query(segs_db);
```
Comment (nemerle) [inline Projects/CoX/Utilities/dbtool/main.cpp]: the old method of initializing the database was a bit better;
```cpp
SOME_CLASS db = SOME_CLASS(arguments);
```
translates into:
```cpp
SOME_CLASS db;
db = SOME_CLASS(arguments);
```
thus we construct a db, and then immediately replace that value with a new one.

**Comment by erpmario** [comment]
Alright, all suggestions have been implemented!

File: Projects/CoX/Servers/AdminServer/AdminDatabase.cpp [hunk 4]
```diff
@@ -92,9 +97,11 @@ bool AdminDatabase::ValidPassword(const char *username, const char *password)
     }
     if(!m_prepared_select_account_passw.next())
         return false;
+    PasswordHasher hasher;
     QByteArray passb = m_prepared_select_account_passw.value("passw").toByteArray();
-    assert(passb.size()<=16);
-    if (memcmp(passb.data(),password,passb.size()) == 0)
+    QByteArray salt = m_prepared_select_account_passw.value("salt").toByteArray();
+    QByteArray hashed_password = hasher.hashPassword(password, salt);
+    if (memcmp(passb.data(),hashed_password.data(),passb.size()) == 0)
```
Comment (nemerle) [inline Projects/CoX/Servers/AdminServer/AdminDatabase.cpp]: I think we can rely on QByteArray comparison here, code will be a bit cleaner

**Pull request merged**